src/java.base/share/classes/sun/security/util/DerValue.java

```diff
@@ -315,6 +318,9 @@ public DerValue(byte[] encoding) throws IOException {
         }
         int pos = offset;
         tag = buf[pos++];
+        if ((tag & 0x1f) == 0x1f) {
+            throw new IOException("Tag number over 30 is not supported");
```

Comment:
Would it be useful for these types of exception messages to either display the offending tag value or perhaps the tag offset?  Just thinking it might be a nice thing for the recipient to know where in the DER encoding the issue is.

Reply:
I don't want to go on reading the following bytes to find out what the intended tag number is, because that somehow shows I do understand the encoding _a lot_ but still don't want to support it (well, actually I only understand _a little_). There are only 2 kinds of tags: one <= 30 and one >= 31. IMHO, the message has already expressed the meaning that we only support the 1st one.

An alternative message I can think of is "Unsupported tag byte: 0xBF", but it looks too cryptic.

Reply:
I think that is fair.  If you don't want to read ahead like that, what about using the "offset" or "pos" field to give a message like "Tag number over 30 at offset <NN> is not supported" (something like that, at least)  Maybe don't worry about the tag value itself, but at least the position in the data stream.  Just a suggestion only, no strong feelings about this either way.

Reply:
There _is_ an `offset` value here but I have really no idea if the user knows where to count from. If we say "offset" then we need to tell what data block we are talking about. What if the DerValue is just a portion of a bigger data block?

That said, if you really like it, I can add an offset like "tag byte at offset nnnn". I like hope the user can find it.